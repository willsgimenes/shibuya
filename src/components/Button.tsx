import React, { FC, ButtonHTMLAttributes } from 'react';
import styled from 'styled-components';
import '../main.css';

const StyledButton = styled.button.attrs(
  ({ variant }: { variant: ButtonVariants }) => ({
    className: `px-8 py-2 font-semibold ${
      variant === 'default' ? 'text-white' : 'text-red-700'
    } transition duration-500 ease-in-out transform rounded-lg shadow-xl bg-gradient-to-r from-red-300 to-blue-300 hover:from-pink-400 hover:to-indigo-400`,
  })
)<{ variant: ButtonVariants }>``;

type ButtonVariants = 'default' | 'warning';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  variant?: ButtonVariants;
}

export const Button: FC<ButtonProps> = ({
  text,
  variant = 'default',
  ...props
}) => {
  return (
    <StyledButton variant={variant} {...props}>
      {text}
    </StyledButton>
  );
};
