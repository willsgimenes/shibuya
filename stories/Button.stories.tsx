import React from 'react';
import { Meta, Story } from "@storybook/react";
import { Button, ButtonProps } from '../src/components/Button';

const meta: Meta = {
    title: 'Button',
    component: Button,
    argTypes: {
        text: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expanded: true }
    }
}

export default meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const PrimaryButton = Template.bind({});
PrimaryButton.args = { text: 'Button', variant: 'primary' };

export const SecondaryButton = Template.bind({});
SecondaryButton.args = { text: 'Button', variant: 'secondary' }

export const TertitaryButton = Template.bind({});
TertitaryButton.args = { text: 'Button', variant: 'tertiary' }
