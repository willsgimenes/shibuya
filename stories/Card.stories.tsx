import React from "react";
import { Meta, Story } from "@storybook/react";
import { Card, CardProps } from "../src/components/Card";

const meta: Meta = {
    title: 'Card',
    component: Card,
}

export default meta;

const content = <div className='p-4'><p>渋谷区</p></div>

const Template: Story<CardProps> = (args) => <Card {...args} />

export const DefaultCard = Template.bind({});

DefaultCard.args = { children: content }
