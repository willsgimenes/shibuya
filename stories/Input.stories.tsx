import React, { useState, useRef } from 'react';
import { Meta, Story } from "@storybook/react";
import { Input, InputProps } from '../src/components/Input';

const meta: Meta = {
    title: 'Input',
    component: Input,
    argTypes: {
        value: {
            control: {
                type: 'text'
            }
        }
    },
    parameters: {
        controls: { expdanded: true }
    }
}

export default meta;

const Template: Story<InputProps> = (args) => React.createElement(() => {
    const [state, setState] = useState(args.value)
    const inputRef = useRef();

    return <Input
        innerRef={inputRef}
        value={state}
        label={args.label}
        onChange={({ target: { value } }) => setState(value)}
        placeholder={args.placeholder}
        disabled={args.disabled}
    />
})

export const DefaultInput = Template.bind({});
DefaultInput.args = { value: 'Yokohama Blues', placeholder: 'Insert text', disabled: false, label: 'song' }
