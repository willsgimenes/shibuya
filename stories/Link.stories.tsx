import React from "react";
import { Meta, Story } from "@storybook/react";
import { Link as SCLink, LinkProps } from '../src/components/Link';

const meta: Meta = {
    title: 'Link',
    component: SCLink,
}

export default meta;

const Template: Story<LinkProps> = (args) => (
    <SCLink {...args}>{args.children}</SCLink>
)

export const Link = Template.bind({});
Link.args = {
    url: "https://open.spotify.com/track/1EJOWPgFH4ThdIFdvbKA2B?si=Zdo4k4UbRnShLeKTHtDr6Q",
    external: true,
    children: 'Listen Yokohama blues',
}