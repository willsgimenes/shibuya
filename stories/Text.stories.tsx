import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Text, TextProps } from "../src/components/Text";

const meta: Meta = {
    title: 'Typography',
    component: Text,
}

export default meta;

const Template: Story<TextProps> = () => <div>
    <Text variant="h1">Heading 1</Text>
    <Text variant="h2">Heading 2</Text>
    <Text variant="h3">Heading 3</Text>
    <Text variant="h4">Heading 4</Text>
    <Text variant="h5">Heading 5</Text>
    <Text variant="p">texto</Text>
    <Text variant='small'>small text</Text>
</div>

export const DefaultTypography = Template.bind({})