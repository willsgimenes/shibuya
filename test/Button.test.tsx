import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { PrimaryButton as Button } from '../stories/Button.stories';

describe('Button', () => {
  it('should render the button without crashing', () => {
    const { getByRole } = render(<Button text="test" />);
    expect(getByRole('button')).toBeInTheDocument();
  });

  it('should call the onClick method when a user clicks on the button', () => {
    const mockedClick = jest.fn();

    const { getByRole } = render(<Button text="test" onClick={mockedClick} />);

    const buttonElement = getByRole('button');

    fireEvent.click(buttonElement);
    expect(mockedClick).toHaveBeenCalledTimes(1);
  });

  it('should have right text color based on variant', () => {
    const { getByRole, rerender } = render(<Button text="test" />);

    const buttonElement = getByRole('button', { name: 'test' });

    expect(buttonElement.classList.contains('bg-purple-800')).toBe(true);
    expect(buttonElement.classList.contains('bg-red-800')).toBe(false);

    rerender(<Button text="test" variant="secondary" />);

    expect(buttonElement.classList.contains('bg-purple-800')).toBe(false);
    expect(buttonElement.classList.contains('bg-red-800')).toBe(true);
  });

  it('should have correct disabled style', () => {
    const { getByRole } = render(<Button text="Button Disabled" disabled />);

    const buttonElement = getByRole('button', { name: 'Button Disabled' });

    expect(buttonElement).toBeDisabled();
  });
});
