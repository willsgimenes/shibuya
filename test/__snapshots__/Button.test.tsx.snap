// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have correct inverted styled 1`] = `
<button
  class="sc-bdfBwQ px-8 py-2 rounded-lg bg-white text-black"
>
  Button Inverted
</button>
`;
